internal/controllers/machine/drain/drain.go

```diff
@@ -0,0 +1,353 @@
+			if ok := errors.As(err, &statusError); ok {
+				errorMessage := statusError.Status().Message
+				if statusError.Status().Details != nil {
+					var causes []string
+					for _, cause := range statusError.Status().Details.Causes {
+						causes = append(causes, cause.Message)
+					}
+					errorMessage = fmt.Sprintf("%s %v", errorMessage, strings.Join(causes, ","))
+				}
+				err = errors.New(errorMessage)
+			}
+
+			log.Error(err, "Error when evicting Pod")
+			res.PodsFailedEviction[err.Error()] = append(res.PodsFailedEviction[err.Error()], pd.Pod)
+		case apierrors.IsForbidden(err) && apierrors.HasStatusCause(err, corev1.NamespaceTerminatingCause):
+			// Creating an eviction resource in a terminating namespace will throw a forbidden error, e.g.:
+			// "pods "pod-6-to-trigger-eviction-namespace-terminating" is forbidden: unable to create new content in namespace test-namespace because it is being terminated"
+			// The kube-controller-manager is supposed to set the deletionTimestamp on the Pod and then this error will go away.
+			msg := "Cannot evict pod from terminating namespace: unable to create eviction (kube-controller-manager should set deletionTimestamp)"
+			log.Error(err, msg)
+			res.PodsFailedEviction[msg] = append(res.PodsFailedEviction[msg], pd.Pod)
+		default:
+			log.Error(err, "Error when evicting Pod")
+			res.PodsFailedEviction[err.Error()] = append(res.PodsFailedEviction[err.Error()], pd.Pod)
+		}
+	}
+
+	return res
+}
+
+// evictPod evicts the given Pod, or return an error if it couldn't.
+func (d *Helper) evictPod(ctx context.Context, pod *corev1.Pod) error {
+	delOpts := metav1.DeleteOptions{}
+	if d.GracePeriodSeconds >= 0 {
+		gracePeriodSeconds := int64(d.GracePeriodSeconds)
+		delOpts.GracePeriodSeconds = &gracePeriodSeconds
+	}
+
+	eviction := &policyv1.Eviction{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      pod.Name,
+			Namespace: pod.Namespace,
+		},
+		DeleteOptions: &delOpts,
+	}
+
+	return d.Client.SubResource("eviction").Create(ctx, pod, eviction)
+}
+
+// EvictionResult contains the results of an eviction.
+type EvictionResult struct {
+	PodsDeletionTimestampSet []*corev1.Pod
+	PodsFailedEviction       map[string][]*corev1.Pod
+	PodsNotFound             []*corev1.Pod
+	PodsIgnored              []*corev1.Pod
+}
+
+// DrainCompleted returns if a Node is entirely drained, i.e. if all relevant Pods have gone away.
+func (r EvictionResult) DrainCompleted() bool {
+	return len(r.PodsDeletionTimestampSet) == 0 && len(r.PodsFailedEviction) == 0
+}
+
+// ConditionMessage returns a condition message for the case where a drain is not completed.
+func (r EvictionResult) ConditionMessage() string {
+	if r.DrainCompleted() {
+		return ""
+	}
+
+	conditionMessage := "Drain not completed yet:"
+	if len(r.PodsDeletionTimestampSet) > 0 {
+		conditionMessage = fmt.Sprintf("%s\n* Pods with deletionTimestamp that still exist: %s", conditionMessage, stringPodList(r.PodsDeletionTimestampSet, 5))
```

Comment:
```suggestion
		conditionMessage = fmt.Sprintf("%s\n* Pods still deleting: %s", conditionMessage, stringPodList(r.PodsDeletionTimestampSet, 5))
```
might be

Reply:
I tried to be very clear to leave minimal room for misunderstandings